tests/test_download.py

```diff
@@ -0,0 +1,89 @@
+import unittest
+from unittest.mock import patch, mock_open
+from pyfakefs.fake_filesystem_unittest import TestCase
+from pathlib import Path, PosixPath
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+
+from gnssanalysis.gn_utils import delete_entire_directory
+import gnssanalysis.gn_download as ga_download
+
+
+class TestDownload(TestCase):
+    def setUp(self):
+        # Create directory to save files
+        self.test_dir = "download_test_dir"
+        Path.mkdir(self.test_dir, exist_ok=True)
+        # self.setUpPyfakefs()
+
+    def tearDown(self):
+        # Clean up test directory after tests:
+        if Path(self.test_dir).is_dir():
+            delete_entire_directory(Path(self.test_dir))
+
+    def test_download_product_from_cddis(self):
+
+        # Test download of file from CDDIS (in this case an IGS ULT)
+        ga_download.download_product_from_cddis(
+            download_dir=Path(self.test_dir),
+            start_epoch=datetime(2024, 8, 10),
+            end_epoch=datetime(2024, 8, 12),
+            file_ext="SP3",
+        )
+
+        # Verify
+        self.assertEqual(type(next(Path(self.test_dir).glob("*.SP3"))), PosixPath)
+        self.assertEqual(next(Path(self.test_dir).glob("*.SP3")).name, "IGS0OPSULT_20242230000_02D_15M_ORB.SP3")
```

Comment:
If we're primarily interested in it determining the **correct** file to download, I'd suggest shifting the focus to the functions that construct the name, rather than the actual download function (that might require some refactoring if they currently call the download itself).

Alternatively if we do download a specific file, I guess we could also hash it (probably doesn't add much benefit to the test though).

Reply:
For this one, yeah we can definitely have a check on the file size for example, but the other files (ATX, metadata, yaw files) will change from time to time, so not sure what else we can check =

Reply:
> For this one, yeah we can definitely have a check on the file size for example, but the other files (ATX, metadata, yaw files) will change from time to time, so not sure what else we can check =

Maybe checksum? Or size + last modified time?